tests/adts/distinct_same_constructor.smt2

```diff
@@ -0,0 +1,9 @@
+(set-logic ALL)
+
+(declare-datatype list
+  ((cons (value Int) (tail list)) (nil)))
+
+(declare-fun l () list)
+(assert (distinct l (cons 0 nil) nil))
```

Comment:
I am not sure I understand what this is trying to test, can you explain?

Reply:
Ok, this test is wrong. I want to write
```smt2
(set-logic ALL)

(declare-datatype list
  ((cons (value Int) (tail list)) (nil)))

(declare-fun l () list)
(assert (distinct l (cons 0 nil)))
(assert (distinct l nil))
```

Reply:
I'm confused. It was correct but this form is clearer.